wandb/data_types.py

```diff
@@ -440,6 +440,21 @@ def to_json(self, run):
             json_obj["_type"] = self.artifact_type
         return json_obj
 
+    @classmethod
+    def from_json(cls, json_obj, source_artifact):
+        """Likely will need to override for any more complicated media objects"""
+        return cls(source_artifact.get_path(json_obj["path"]).download())
+
+    def __eq__(self, other):
+        """Likely will need to override for any more complicated media objects"""
+        return (
+            self.__class__ == other.__class__
+            and hasattr(self, "_sha256")
+            and hasattr(other, "_sha256")
+            and self._sha256 == other._sha256
+            and self._size == other._size
```

Comment:
What's the case where the sha256's are the same and the size is different?

Reply:
Ah - no case. I removed this in the next commit